src/Plugin/yaml_content/process/ProcessFile.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace Drupal\tide_demo_content\Plugin\yaml_content\process;
+
+use Drupal\Component\Render\PlainTextOutput;
+use Drupal\yaml_content\Plugin\ProcessingContext;
+use Drupal\yaml_content\Plugin\yaml_content\process\File;
+
+/**
+ * Plugin for processing and loading a file attachment.
+ *
+ * @YamlContentProcess(
+ *   id = "file_path_rename",
+ *   title = @Translation("File Processor"),
+ *   description = @Translation("Processing and loading a file attachment.")
+ * )
+ */
+class ProcessFile extends File {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function process(ProcessingContext $context, array &$field_data) {
+    $entity_type = $this->configuration[0];
+    $filter_params = $this->configuration[1];
+
+    $filename = $filter_params['filename'];
+    $directory = '/data_files/';
+    // If the entity type is an image, look in to the /images directory.
+    if ($entity_type == 'image') {
+      $directory = '/images/';
+    }
+    // TODO path not set.
```

Comment:
@MdNadimHossain Is this comment still relevant?

Reply:
@GROwen I missed on that, the comment doesn't need to be there. As the path is set to the custom path, I will remove this comment.